spec/use_cases/administrator/validate_log_search_query_spec.rb

```diff
@@ -0,0 +1,35 @@
+describe UseCases::Administrator::ValidateLogSearchQuery do
+  let(:email) { 'test@example.com' }
+
+  context 'Valid search params' do
+    context 'valid username' do
+      it 'returns true' do
+        valid_params = [
+          { username: 'ABCDEF', ip: nil },
```

Comment:
Can I search for both at once?

Reply:
The UI won't allow you to search both at once.
You will either search username by input field, or ip by link from the settings page.